packages/php-domain/src/PhpDomain/MethodNameService.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare(strict_types=1);
+
+namespace T3Docs\PhpDomain\PhpDomain;
+
+use T3Docs\PhpDomain\Nodes\MethodNameNode;
+
+class MethodNameService
+{
+    private const PATTERN = '/^\s*(\w+)\s*\(\s*(.*?)\s*\)\s*(?::\s*(\w+))?\s*$/';
```

Comment:
Please provide a regex101.com link and provide there some examples. This makes it easier to get what the regex does.

There is also a phpstan rule which enforces this in the package symplify/php-stanrules:
https://github.com/symplify/phpstan-rules/blob/main/docs/rules_overview.md#annotateregexclassconstwithregexlinkrule
Maybe we should add this.

Reply:
Yes, why not. Would you like to make a PR for that rule?

Reply:
I added the comment